src/libraries/System.Net.Http/src/System/Net/Http/MultipartContent.cs

```diff
@@ -669,6 +681,35 @@ private static void ValidateReadArgs(byte[] buffer, int offset, int count)
             public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken) { throw new NotSupportedException(); }
             public override ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationToken cancellationToken = default) { throw new NotSupportedException(); }
         }
+
+
+        private static void WriteToStream(Stream stream, string content, Encoding encoding)
+        {
+            if (ReferenceEquals(encoding, Encoding.UTF8) && content.Length <= 128)
+            {
+                // Fast-path for the expected common case
```

Comment:
isn't Latin1/Ascii even more common path? It can bypass encoding.GetBytes at all and directly copy string char by char into the stream. 

`WriteLatin1ToStream` would be `WriteToStream(stream, content, Encoding.Latin1)`

Reply:
The default behavior (not setting the selector / returning null for encoding) is handeled in `WriteLatin1ToStream`.

This method (`WriteToStream`) will only be called if a custom encoding is supplied. We can do extra checks here in case they manually pass `Latin1` as well.

> It can bypass encoding.GetBytes at all and directly copy string char by char into the stream.

Latin1/Ascii encodings aren't just blind copy operations, they will try to apply best-fit logic on characters outside their range. The implementations are also vectorized, so faster overall than a manual loop would be.